src/main/java/factionsplusplus/repositories/LockedBlockRepository.java

```diff
@@ -1,65 +1,44 @@
 package factionsplusplus.repositories;
 
 import com.google.inject.Singleton;
-
-import java.io.File;
-import java.io.FileInputStream;
-import java.io.InputStreamReader;
-import java.io.OutputStreamWriter;
-import java.nio.file.Files;
-import java.nio.charset.StandardCharsets;
-
-import com.google.gson.Gson;
-import com.google.gson.GsonBuilder;
-import com.google.gson.stream.JsonReader;
 import com.google.inject.Inject;
-import com.google.inject.name.Named;
-import com.google.gson.reflect.TypeToken;
-
-import java.io.IOException;
-import java.io.FileNotFoundException;
 
 import org.bukkit.block.Block;
 
 import java.util.List;
 import java.util.UUID;
 import java.util.ArrayList;
-import java.lang.reflect.Type;
 
+import factionsplusplus.data.LockedBlockDao;
 import factionsplusplus.models.LockedBlock;
+import factionsplusplus.services.DataProviderService;
 import factionsplusplus.utils.Logger;
 
 @Singleton
 public class LockedBlockRepository {
     private List<LockedBlock> lockedBlockStore = new ArrayList<>();
-    private final String dataPath;
-    private final static String FILE_NAME = "lockedblocks.json";
-    private final static Type JSON_TYPE = new TypeToken<List<LockedBlock>>() { }.getType();
     private final Logger logger;
+    private final DataProviderService dataProviderService;
 
     @Inject
-    public LockedBlockRepository(@Named("dataFolder") String dataPath, Logger logger) {
-        this.dataPath = String.format("%s%s%s", dataPath, File.separator, FILE_NAME);
+    public LockedBlockRepository(Logger logger, DataProviderService dataProviderService) {
         this.logger = logger;
+        this.dataProviderService = dataProviderService;
     }
 
     // Load claimed chunks
     public void load() {
-        this.lockedBlockStore.clear();
         try {
-            Gson gson = new GsonBuilder()
-                .excludeFieldsWithoutExposeAnnotation()
-                .serializeNulls()
-                .create();
-            JsonReader reader = new JsonReader(new InputStreamReader(new FileInputStream(this.dataPath), StandardCharsets.UTF_8));
-            this.lockedBlockStore = gson.fromJson(reader, LockedBlockRepository.JSON_TYPE);
-        } catch (FileNotFoundException ignored) {
-            this.logger.error(String.format("File %s not found", this.dataPath), ignored);
+            this.lockedBlockStore.clear();
+            this.lockedBlockStore = this.getDAO().getAll();
+        } catch(Exception e) {
+            this.logger.log(String.format("Error loading locked blocks: %s", e.getMessage()));
         }
     }
 
     // Save a locked block
     public void create(LockedBlock block) {
+        this.getDAO().create(block);
```

Comment:
I think you're forgetting to delete from DAO during `delete`. I couldn't comment on that line so i'm writing it here